### PR TITLE
{CI} enable pytest group worker in module test and full test

### PR DESCRIPTION
**Pull request by wangzelin007**
**Description**

- Add --pytest-args "--durations=0" to record the time spent on each test.
- For module test and full test:
  Every module not in serial_modules, enable group worker in pytest, pytest will spawn a number of workers processes equal to the number of available CPUs, and distribute the tests randomly across them.
- For incremental test:
  Remain the same
- serial_modules: ['appservice', 'botservice', 'cloud', 'network', 'azure-cli-core', 'azure-cli-telemetry']

image

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
CI

File: scripts/ci/automation_full_test.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    def get_all_modules(self):
+        result = get_path_table()
+        # only get modules and core, ignore extensions
+        self.modules = {**result['mod'], **result['core']}
+
+    def run_modules(self):
+        # divide all modules into parallel or serial execution
+        error_flag = False
+        series = []
+        parallers = []
```
Comment (jiasli) [inline scripts/ci/automation_full_test.py:37]: The word is "parallel".

File: .azure-pipelines/templates/automation_test.yml [hunk 2]
```diff
@@ -32,23 +32,31 @@ steps:
       echo fullTest: ${{ parameters.fullTest }}
       echo module: ${{ parameters.module }}
       echo Build.Reason: $(Build.Reason)
+      series_modules="appservice botservice cloud network azure-cli-core azure-cli-telemetry"
```
Comment (jiasli) [inline .azure-pipelines/templates/automation_test.yml:35]: `series_modules` are defined twice, one here, one in `automation_full_test.py`. Maybe we can unify them to reduce duplication.

Comment (wangzelin007) [reply .azure-pipelines/templates/automation_test.yml:35]: Modify series_modules to serial_modules to ensure consistency.

**Review by jiasli — APPROVED** [review]

**Review by evelyn-ys — APPROVED** [review]

**Pull request merged**